### ISSUE TITLE
[LTO] Unit test testCudaCheck failing for `aarch64` architecture

### ISSUE THREAD

**Issue by aandvalenzuela** [cms-sw/cmssw#40834]
Hello,

Test `testCudaCheck` (module `HeterogeneousCore/CUDAUtilities`) is failing since 1st of Feb in LTO IBs (`aarch64` architecture only) due to segmentation violation when checking the driver API:
```
/data/cmsbld/jenkins_a/workspace/build-any-ib/w/tmp/BUILDROOT/d82d494993e018306d251480a039b1b4/opt/cmssw/el8_aarch64_gcc11/cms/cmssw/CMSSW_13_1_LTO_X_2023-02-19-0000/src/HeterogeneousCore/CUDAUtilities/test/testCudaCheck.cpp:13: FAILED:
  {Unknown expression after the reported line}
due to a fatal error condition:
  SIGSEGV - Segmentation violation signal

===============================================================================
test cases: 1 | 1 failed
assertions: 2 | 1 passed | 1 failed


 *** Break *** segmentation violation
```
See [stacktrace](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_aarch64_gcc11/CMSSW_13_1_LTO_X_2023-02-20-2300/unitTestLogs/HeterogeneousCore/CUDAUtilities#/333-333).

Test was added on 1st of Feb via https://github.com/cms-sw/cmssw/pull/40619 and it succeeds since then in `amd64`, but not on `aarch64`. Tests are supposed to pass on machines with and without GPUs (https://github.com/cms-sw/cmssw/pull/40619#issue-1558191479). I can reproduce the issue in all our `aarch64` machines.

Thanks,
Andrea.

**Comment by aandvalenzuela** [comment]
assign heterogeneous

**Comment by aandvalenzuela** [comment]
Not sure why it succeeds in non-LTO IBs:
```
===== Test "testCudaCheck" ====
===============================================================================
All tests passed (4 assertions in 1 test case)


---> test testCudaCheck succeeded
```
[CMSSW_13_1_X_2023-02-20-2300](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_aarch64_gcc11/CMSSW_13_1_X_2023-02-20-2300/unitTestLogs/HeterogeneousCore/CUDAUtilities#/261-261)  for `el8_aarch64_gcc11` also.

**Comment by makortel** [comment]
I took a look and it seems that the `error` and `message` in 
https://github.com/cms-sw/cmssw/blob/918a7fcd99447e01c7ba70f6bf5e04fa9a70b080/HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h#L45-L49
stay uninitialized after the `cuGetErrorName()` and `cuGetErrorString()` calls. Ok, I didn't really check that, but their value in the `abortOnCudaError()` are garbage, and initializing them to `nullptr` seems to fix the crash.

**Comment by makortel** [comment]
Fixed in #40840

**Comment by aandvalenzuela** [comment]
Thanks @makortel!
